syntaxes/crunch.lang

```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<language id="crunch" _name="Crunch" version="2.0" _section="Source">
+  <metadata>
+    <property name="globs">*.crunch</property>
+    <property name="line-comment-start">::</property>
+  </metadata>
+
+  <styles>
+    <style id="comment" _name="Comment" map-to="def:comment"/>
+    <style id="string" _name="String" map-to="def:string"/>
+    <style id="char" _name="Character" map-to="def:character"/>
+    <style id="keyword" _name="Keyword" map-to="def:keyword"/>
+    <style id="type" _name="Data Type" map-to="def:type"/>
+    <style id="boolean" _name="Boolean value" map-to="def:boolean"/>
+    <style id="identifier" _name="Identifier" map-to="def:identifier"/>
+    <style id="number" _name="Number" map-to="def:number"/>
+    <style id="scope" _name="Scope" map-to="def:preprocessor"/>
+    <style id="attribute" _name="Attribute" map-to="def:preprocessor"/>
+    <style id="macro" _name="Macro" map-to="def:preprocessor"/>
+  </styles>
+
+  <definitions>
+    <context id="keywords" style-ref="keyword">
+      <keyword>as</keyword>
+      <keyword>break</keyword>
+      <keyword>continue</keyword>
+      <keyword>else</keyword>
+      <keyword>empty</keyword>
+      <keyword>end</keyword>
+      <keyword>enum</keyword>
+      <keyword>exposed</keyword>
+      <keyword>exposing</keyword>
+      <keyword>fn</keyword>
+      <keyword>for</keyword>
+      <keyword>if</keyword>
+      <keyword>import</keyword>
+      <keyword>in</keyword>
+      <keyword>let</keyword>
+      <keyword>loop</keyword>
+      <keyword>match</keyword>
+      <keyword>pkg</keyword>
+      <keyword>return</keyword>
+      <keyword>then</keyword>
+      <keyword>trait</keyword>
+      <keyword>type</keyword>
+      <keyword>where</keyword>
+      <keyword>while</keyword>
+    </context>
+    
+    <!--
+    <context id="reserved" style-ref="def:reserved">
+    </context>
+    -->
+
+    <context id="types" style-ref="type">
+      <match extended="true">
+        \b(bool|unit|int|float|str)\b
```

Comment:
I don't know if it's possible, but also allowing `&` and `&mut` somehow would be great

Reply:
hmm, i think & is distinguishable enough to not need it. the `mut` part *will* get highlighted tho.